### PR TITLE
Content script clarifications

### PR DESCRIPTION
**Pull request by rebloor**
#### Summary
Clarifications of content script behavior in Manifest V2 and V3

#### Metadata
This PR…

- [ ] Adds a new document
- [x] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/mozilla/add-ons/webextensions/content_scripts/index.md [hunk 1]
```diff
@@ -183,6 +183,8 @@ Content scripts can make requests using the normal [`window.XMLHttpRequest`](/en
 
 Content scripts get the same cross-domain privileges as the rest of the extension: so if the extension has requested cross-domain access for a domain using the [`permissions`](/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions) key in `manifest.json`, then its content scripts get access that domain as well.
 
+> **Note:** From Firefox 101, content scripts can perform cross-origin requests when the destination server opts in using [CORS](/en-US/docs/Web/HTTP/CORS); however, host permissions don't work in content scripts, but they still do in regular extension pages.
```
Comment (Rob--W) [inline files/en-us/mozilla/add-ons/webextensions/content_scripts/index.md:186]: This change is tied to MV3, not a Firefox version.

File: files/en-us/mozilla/firefox/releases/101/index.md [hunk 2]
```diff
@@ -72,10 +72,13 @@ The [`prefers-contrast`](/en-US/docs/Web/CSS/@media/prefers-contrast) media feat
...
 - The [`"background"`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/manifest.json/background) manifest key property `"persistent"` can be set to `false` under the control of preferences: for Manifest V2, the <code>extensions.eventPages.enabled</code> preference, and in Manifest V3, the <code>extensions.manifestV3.enabled</code> preference.
-- Addition of the [`"host_permissions"`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/manifest.json/host_permissions) manifest key, which is available for Manifest V3 extensions.
+- Addition of the [`"host_permissions"`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/manifest.json/host_permissions) manifest key, which is available for Manifest V3 extensions. 
+- Content scripts execution has changed for Manifest V3 extensions:
```
Comment (Rob--W) [inline files/en-us/mozilla/firefox/releases/101/index.md:79]:
```suggestion
- The content script execution environment has changed for Manifest V3 extensions:
```

File: files/en-us/mozilla/firefox/releases/101/index.md [hunk 3]
```diff
@@ -72,10 +72,13 @@ The [`prefers-contrast`](/en-US/docs/Web/CSS/@media/prefers-contrast) media feat
...
+  - content scripts can only send requests that require CORS, not host permissions.
```
Comment (Rob--W) [inline files/en-us/mozilla/firefox/releases/101/index.md:80]:
```suggestion
  - Content scripts can no longer rely on host permissions to perform cross-origin requests. Cross-origin requests from content scripts are possible with CORS.
```

The previous statement is incorrect; same-origin requests (i.e. to the same origin as the page) are also/still possible.

**Review by Rob--W — CHANGES_REQUESTED** [review]

File: files/en-us/mozilla/add-ons/webextensions/content_scripts/index.md (hunk 1 above)
Comment (rebloor) [inline files/en-us/mozilla/add-ons/webextensions/content_scripts/index.md:186]:
```suggestion
> **Note:** When using Manifest V3, content scripts can perform cross-origin requests when the destination server opts in using [CORS](/en-US/docs/Web/HTTP/CORS); however, host permissions don't work in content scripts, but they still do in regular extension pages.
```

Comment (rebloor) [reply files/en-us/mozilla/add-ons/webextensions/content_scripts/index.md]: Updated

File: files/en-us/mozilla/add-ons/webextensions/content_scripts/index.md [hunk 4]
```diff
@@ -191,8 +193,8 @@ This is accomplished by exposing more privileged XHR and fetch instances in the
 >
 > This is not possible in Manifest V3, as `content.XMLHttpRequest` and `content.fetch()` are not available.
 
-> **Note:** In Chrome, starting with version 73, and Firefox, starting with version 101 when using Manifest V3, content scripts are subject to the same CORS policy as the page they are running within. Only backend scripts have elevated cross-domain privileges. See [Changes to Cross-Origin Requests in Chrome Extension Content Scripts](https://www.chromium.org/Home/chromium-security/extension-content-script-fetches/).
-
+> **Note:** In Chrome, starting with version 73, and Firefox, starting with version 101 when using Manifest V3, content scripts are subject to the same [CORS](/en-US/docs/Web/HTTP/CORS) policy as the page they are running within. Only backend scripts have elevated cross-domain privileges. See [Changes to Cross-Origin Requests in Chrome Extension Content Scripts](https://www.chromium.org/Home/chromium-security/extension-content-script-fetches).
```
Comment (rebloor) [inline files/en-us/mozilla/add-ons/webextensions/content_scripts/index.md:196]:
```suggestion
> **Note:** In Chrome, starting with version 73, and Firefox, starting with version 101 when using Manifest V3, content scripts are subject to the same [CORS](/en-US/docs/Web/HTTP/CORS) policy as the page they are running within. Only backend scripts have elevated cross-domain privileges. See [Changes to Cross-Origin Requests in Chrome Extension Content Scripts](https://www.chromium.org/Home/chromium-security/extension-content-script-fetches). 
```

File: files/en-us/mozilla/firefox/releases/101/index.md [hunk 5]
```diff
@@ -72,10 +72,13 @@ The [`prefers-contrast`](/en-US/docs/Web/CSS/@media/prefers-contrast) media feat
 
 ## Changes for add-on developers
 
-- Addition of the {{WebExtAPIRef("scripting")}} API, which provides features to execute a script, insert and remove CSS, and manage the registration of content scripts ({{bug(1687764)}}). This API is available to Manifest V3 extensions and takes over the execute script and insert and remove CSS features from the {{WebExtAPIRef("tabs")}} API.
+- Addition of the {{WebExtAPIRef("scripting")}} API, which provides features to execute a script, insert and remove CSS, and manage the registration of content scripts ({{bug(1687764)}}). This API is available to Manifest V3 extensions and takes over the execute script and insert and remove CSS features from the {{WebExtAPIRef("tabs")}} API.
```
Comment (rebloor) [inline files/en-us/mozilla/firefox/releases/101/index.md:75]:
```suggestion
- Addition of the {{WebExtAPIRef("scripting")}} API, which provides features to execute a script, insert and remove CSS, and manage the registration of content scripts ({{bug(1687764)}}). This API is available to Manifest V3 extensions and takes over the execute script and insert and remove CSS features from the {{WebExtAPIRef("tabs")}} API.
```

**Review by Rob--W — APPROVED** [review]
Thanks as usual Richard!

**Pull request merged**